lib/cluster-providers/generic-cluster-provider.ts

```diff
@@ -42,6 +43,85 @@ export interface GenericClusterProviderProps extends eks.ClusterOptions {
+     */
+    minSize = new NumberConstraint(0, 100);
+
+    /**
+     * maxSize has a maximum of 100 nodes per node group and can be as little as 0.
+     * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+     */
+    maxSize = new NumberConstraint(0, 100);
+
+    /**
+     * desiredSize has a maximum of 100 nodes per node group and can be as little as 0.
+     * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+     */
+    desiredSize = new NumberConstraint(0, 100);
+
+    /**
+     * amiReleaseVersion can be no less than 1 character long, and no greater than 63 characters long.
+     * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+     */
+    amiReleaseVersion = new StringConstraint(1, 63);
+}
+
+export class AutoscalingNodeGroupConstraints implements ConstraintsType<AutoscalingNodeGroup> {
+    /**
+    * id can be no less than 1 character long, and no greater than 63 characters long.
+    * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+    */
+    id = new StringConstraint(1, 63);
+
+    /**
+    * minSize has a maximum of 100 nodes per node group and can be as little as 0.
+    * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+    */
+    minSize = new NumberConstraint(0, 100);
+
+    /**
+    * maxSize has a maximum of 100 nodes per node group and can be as little as 0.
+    * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+    */
+    maxSize = new NumberConstraint(0, 100);
+
+    /**
+    * desiredSize has a maximum of 100 nodes per node group and can be as little as 0.
+    * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+    */
+    desiredSize = new NumberConstraint(0, 100);
+}
+
+export class FargateProfileConstraints implements ConstraintsType<FargateProfileOptions> {
+    /**
+     * fargateProfileNames can be no less than 1 character long, and no greater than 63 characters long.
+     * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+     */
+    fargateProfileName = new StringConstraint(1, 63);
+}
+
+export class GenericClusterPropsConstraints implements ConstraintsType<GenericClusterProviderProps> {
+    /**
+     * managedNodeGroups max size is 10 managed node groups per EKS cluster, and as little as 0.
+     * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster
```

Comment:
This is dated: 18 NOV 2019
Things changed since then. Only use official AWS documentation for reference. E.g. we can have by default 450 nodes per node group. And even that is a soft limit. I would relax this constraint and use a hard limit. Needs more research.

Reply:
I believe this link should suffice: https://docs.aws.amazon.com/eks/latest/userguide/service-quotas.html

And I would add a comment along the lines of "managedNodeGroups soft limit is 450 managed node groups per EKS cluster, and as little as 0. But we multiply that by a factor of 5 to 2250 for the situations of hard limit requests made"

Reply:
While the examples of nodes per managed node group would be: 
"nodes per node group has a soft limit of 450 node, and as little as 0. But we multiply that by a factor of 5 to 2250 in case of situations of a hard limit request being accepted, and as a result the limit would be raised".